src/material/list/testing/list-harness.ts

```diff
@@ -0,0 +1,299 @@
+  }
+
+  /**
+   * Gets a list of harnesses representing all of the items, subheaders, and dividers
+   * (in the order they appear in the list). Use `instanceof` to check which type of harness a given
+   * item is.
+   * @param filters Optional filters used to narrow which list items, subheaders, and dividers are
+   *     included. A value of `false` for the `item`, `subheader`, or `divider` properties indicates
+   *     that the respective harness type should be omitted completely.
+   * @return The list of harnesses representing the items, subheaders, and dividers matching the
+   *     given filters.
+   */
+  getItemsSubheadersAndDividers(filters: {
+    item: false,
+    subheader: false,
+    divider: false
+  }): Promise<[]>;
+  getItemsSubheadersAndDividers(filters: {
+    item?: F | false,
+    subheader: false,
+    divider: false
+  }): Promise<C[]>;
+  getItemsSubheadersAndDividers(filters: {
+    item: false,
+    subheader?: SubheaderHarnessFilters | false,
+    divider: false
+  }): Promise<MatSubheaderHarness[]>;
+  getItemsSubheadersAndDividers(filters: {
+    item: false,
+    subheader: false,
+    divider?: DividerHarnessFilters | false
+  }): Promise<MatDividerHarness[]>;
+  getItemsSubheadersAndDividers(filters: {
+    item?: F | false,
+    subheader?: SubheaderHarnessFilters | false,
+    divider: false
+  }): Promise<(C | MatSubheaderHarness)[]>;
+  getItemsSubheadersAndDividers(filters: {
+    item?: F | false,
+    subheader: false,
+    divider?: false | DividerHarnessFilters
+  }): Promise<(C | MatDividerHarness)[]>;
+  getItemsSubheadersAndDividers(filters: {
+    item: false,
+    subheader?: false | SubheaderHarnessFilters,
+    divider?: false | DividerHarnessFilters
+  }): Promise<(MatSubheaderHarness | MatDividerHarness)[]>;
+  getItemsSubheadersAndDividers(filters?: {
+    item?: F | false,
+    subheader?: SubheaderHarnessFilters | false,
+    divider?: DividerHarnessFilters | false
+  }): Promise<(C | MatSubheaderHarness | MatDividerHarness)[]>;
+  async getItemsSubheadersAndDividers(filters: {
+    item?: F | false,
+    subheader?: SubheaderHarnessFilters | false,
+    divider?: DividerHarnessFilters | false
+  } = {}): Promise<(C | MatSubheaderHarness | MatDividerHarness)[]> {
+    const query = [];
+    if (filters.item !== false) {
+      query.push(this._itemHarness.with(filters.item || {} as F));
+    }
+    if (filters.subheader !== false) {
+      query.push(MatSubheaderHarness.with(filters.subheader));
+    }
+    if (filters.divider !== false) {
+      query.push(MatDividerHarness.with(filters.divider));
+    }
+    return this.locatorForAll(...query)();
+  }
+}
+
+/** Harness for interacting with a standard mat-list in tests. */
+export class MatListHarness extends
+    MatListHarnessBase<typeof MatListItemHarness, MatListItemHarness, ListItemHarnessFilters> {
+  /** The selector for the host element of a `MatList` instance. */
+  static hostSelector = MAT_LIST_SELECTOR;
```

Comment:
Instead of making a separate constant, would it make sense to reference this elsewhere as `MatListHarness.hostSelector`?
(similar for other selectors)

Reply:
That's what I wanted to do originally, but it would introduce a circular dependency.